### DISCUSSION TITLE
[BP#106] Désactiver les logs binaires de MySQL / MariaDb

### DISCUSSION THREAD

**Discussion by devcrafting** [cnumr/best-practices#217] [Bonnes pratiques] [answered]
Je ne suis pas sûr de voir ce qu'on perd en désactivant ces logs? Ce n'est pas le transaction log qui peut servir de backup incremental?

Est-ce spécifique à MySQL/MariaDB?

**Comment by ACTLEM** [comment]
En phase. Je suis donc plus pour remplacer par une BP sur la mise en place d'un log rotate, durée de rétention de logs sur les serveurs + bdd.

**Comment by ldesgrange** [comment, marked as answer]
Les log binaires sont utilisés pour la réplication vers un deuxième (ou plus) autres serveurs en presque temps réel. Le but est d'avoir de pouvoir faire du failover pour de la haute disponibilité et un backup temps-réel.

Ils peuvent être aussi utilisés pour réappliquer les derniers changements dans la base lors de la restauration d'un backup.

Si la réplication n'est pas utilisée et que la perte des données depuis le dernier backup est acceptable, supprimer les logs binaires est une bonne idée (a minima dans les environnements de dev/test/staging).

Si les logs binaires sont utilsés, il y a des options pour limiter leur taille / durée.

**Reply by ACTLEM** [reply]
Effectivement, ici c'est le terme "log binaire" qui porte à confusion. Comme le disait @devcrafting, on parle plutôt de transaction.log (ou journal de transaction en français).
A mon avis, il y a donc une modification à faire sur le wording et avoir un disclaimer sur le fait de s'assurer que désactiver doit s'inscrire dans une politique de sauvegarde.

Pour les logs "applicatifs", je serais pour créer une BP sur la rétention, log rotate.

**Reply by devcrafting** [reply]
Je ne connais pas bien MySQL, mais en SQL Server il y a la possibilité d'être dans un mode (plutôt recommandé) de faire des backups de logs avec backups full qui tronque les transactions logs. La réplication fonctionne aussi sur cette base de transaction log (mais un doute sur le fonctionnement de tronquer les logs au backup avec la réplication). 
Bref je dirais que l'essence de cette BP c'est faire en sorte d'avoir une rotation des transactions logs (si nécessaire => lien procédure de backup, avertissement à mettre en effet) => titre "surveiller/gérer/contrôler ses logs de transactions"

+1 sur la remarque sur les logs applicatives (il n'y a pas une BP au sujet de ces logs?)

**Reply by ACTLEM** [reply]
Si, il y a la [BP#103](https://github.com/cnumr/best-practices/blob/main/chapters/BP_103_fr.md), mais à améliorer pour intégrer tous les serveurs du service web (serveur applicatif, base de donnée, http...), je pense.

**Reply by ACTLEM** [reply]
Issue créée pour amélioration de la description, du wording: https://github.com/cnumr/best-practices/issues/290
